soc/xtensa/intel_adsp/common/boot.c

```diff
@@ -66,6 +66,8 @@
 
 #define MANIFEST_SEGMENT_COUNT 3
 
+extern void soc_trace_init(void);
```

Comment:
The extern keyword on a function declaration is ignored by the compiler and can be safely omitted.

Reply:
Indeed.  It's a really, really common idiom in the kernel though, so it's in my fingers.  I guess I've grown to like it for uses like this ("declare a function no one put in a header").  Though in this particular case it's an artifact of the fact that this bootloader code used to be built entirely separately and didn't share headers.  At this point there's no reason not to just move this to soc.h.

Reply:
Harbison and Steele 4.8.5 recommends the use of `extern` to be compatible with the largest number of toolchains. Is there some documented Zephyr policy about this? I know Zephyr has a complex CMake toolchain abstraction layer which seems aimed at supporting a wide range of toolchains, possibly some "less standard" ones.

Reply:
If you could move this to a header then that would be great and this discussion is moot.

To continue this discussion, about what should be policy, and not about this particular instance could be interesting.

That is interesting @march-hb. I was not aware that toolchains can have trouble with this. But I have also never seen any patch to this effect, adding extern to fix a toolchain issue, which one would assume would have happened by now if any third-party toolchain building Zephyr did have this problem. So I believe this is not an issue, but time may prove me wrong ...

IMHO it should be policy to not use it.

Assuming I am right and that no toolchain building Zephyr will ever have a problem with this, then it acts only as documentation to the reader, which IMHO would be better served with a comment.

// We declare this external function here instead of including it's header becase ...